source/extensions/tracers/opentelemetry/resource_detectors/environment/environment_resource_detector.cc

```diff
@@ -0,0 +1,57 @@
+#include "environment_resource_detector.h"
+
+#include <sstream>
+#include <string>
+
+#include "source/common/config/datasource.h"
+#include "source/extensions/tracers/opentelemetry/resource_detectors/resource_detector.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+constexpr absl::string_view kOtelResourceAttributesEnv = "OTEL_RESOURCE_ATTRIBUTES";
+
+/**
+ * @brief Detects a resource from the OTEL_RESOURCE_ATTRIBUTES environment variable
+ * Based on the OTel C++ SDK:
+ * https://github.com/open-telemetry/opentelemetry-cpp/blob/v1.11.0/sdk/src/resource/resource_detector.cc
+ *
+ * @return Resource A resource with the attributes from the OTEL_RESOURCE_ATTRIBUTES environment
+ * variable.
+ */
+Resource EnvironmentResourceDetector::detect() {
+  envoy::config::core::v3::DataSource ds;
+  ds.set_environment_variable(kOtelResourceAttributesEnv);
+
+  Resource resource;
+  resource.schemaUrl_ = "";
+  std::string attributes_str = "";
+
+  TRY_NEEDS_AUDIT {
+    attributes_str = Config::DataSource::read(ds, true, context_.serverFactoryContext().api());
+  }
+  END_TRY catch (const EnvoyException& e) {
+    ENVOY_LOG(warn, "Failed to detect resource attributes from the environment: {}.", e.what());
+  }
+
+  if (attributes_str.empty()) {
+    return resource;
+  }
+
+  std::istringstream iss(attributes_str);
+  std::string token;
+  while (std::getline(iss, token, ',')) {
+    size_t pos = token.find('=');
```

Comment:
Done in: [96d2bc6](https://github.com/envoyproxy/envoy/pull/29547/commits/96d2bc6683f8d850e7d211af6ec1c4b0e0fdc433) (add a check, bit different now because of I'm using `splitToken`)